### PR TITLE
 feat: fetch all billing plans

### PR DESCRIPTION
**Pull request by Ayobamidele**
## Description

This pull request introduces a new endpoint `GET /api/v1/organizations/billing-plans` that allows users to retrieve all billing plans associated with a specified organization. The endpoint responds with a JSON object containing the status of the request, a message, and an array of billing plans, each with its ID, name, price, and features.

### Example Response:

```json
{
  "status": 200,
  "message": "plans fetched successfully",
  "data": {
    "plans": [
      {
        "id": "basic",
        "name": "Basic Plan",
        "price": 10.00,
        "features": ["Feature 1", "Feature 2"]
      },
      {
        "id": "premium",
        "name": "Premium Plan",
        "price": 20.00,
        "features": ["Feature 1", "Feature 2", "Feature 3"]
      }
    ]
  }
}
```
Related Issue (Link to issue ticket)
[#321](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/321)

**Motivation and Context**

This change is required to provide organizations with the ability to view available billing plans. It addresses the need for enhanced visibility of billing options, allowing organizations to make informed decisions about subscription levels and features they may want to use.

**How Has This Been Tested?**

The new endpoint has been tested using Postman to ensure that the correct billing plans are fetched based on valid organization IDs. Testing included the following:

    

- [x] Sending requests with valid authorization tokens.
- [x] Verifying the response format, status code, and data integrity.
- [x] Ensuring that appropriate status codes are returned for unauthorized tokens

Screenshot 2024-07-24 125004
Screenshot 2024-07-24 125025

Types of changes

   - [ ] Bug fix (non-breaking change which fixes an issue)
   - [x] New feature (non-breaking change which adds functionality)
   - [ ] Breaking change (fix or feature that would cause existing functionality to change)

Checklist:

  -  [x] My code follows the code style of this project.
  -  [ ] My change requires a change to the documentation.
  -  [ ] I have updated the documentation accordingly.
  -  [x] I have read the CONTRIBUTING document.
  -  [x] I have added tests to cover my changes.
  -  [x] All new and existing tests passed.

### CODE REVIEW

**Review by Goketech — CHANGES_REQUESTED** [review]
make changes to your pr title to follow guidelines

use consistent naming

change plan.py in services to billing_plan.py

**Comment by buka4rill** [comment]
Rename PR from Feat to feat

**Comment by Ayobamidele** [comment]
@Goketech Please review

**Comment by Ayobamidele** [comment]
@zxenonx I've made the changes

**Review by Goketech — APPROVED** [review]

**Comment by bolexs** [comment]
resolve ur conflict

**Comment by Ayobamidele** [comment]
@bolexs It's done sir

**Review by Goketech — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

**Review by Goketech — DISMISSED** [review]

File: tests/v1/test_billing_plan.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+BILLPLAN_ENDPOINT = '/api/v1/organizations/billing-plans'
+
+
+@pytest.fixture
+def mock_db_session():
+    """Fixture to create a mock database session."""
+
+    with patch("api.v1.services.user.get_db", autospec=True) as mock_get_db:
+        mock_db = MagicMock()
+        # mock_get_db.return_value.__enter__.return_value = mock_db
```
Comment (buka4rill) [inline tests/v1/test_billing_plan.py:29]: take out unnecessary comment

File: requirements.txt [hunk 2]
```diff
@@ -52,7 +52,7 @@ typer==0.12.3
 typing_extensions==4.12.2
 uuid7==0.1.0
 uvicorn==0.30.3
-uvloop==0.19.0
+# uvloop==0.19.0
```
Comment (buka4rill) [inline requirements.txt:55]: unecessary comment

**Review by buka4rill — COMMENTED** [review]
some unnecessary comments

**Review by Goketech — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

**Review by zxenonx — APPROVED** [review]
Thanks @Ayobamidele

**Comment by zxenonx** [comment]
Please resolve the merge conflicts @Ayobamidele

**Comment by Goketech** [comment]
tests are failing

**Comment by Ayobamidele** [comment]
@zxenonx @Goketech @joboy-dev I've done it sir

**Review by joboy-dev — APPROVED** [review]

**Review by Goketech — APPROVED** [review]

**Review by zxenonx — APPROVED** [review]
Thanks @Ayobamidele

**Pull request merged**